src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/request/JsonTemplateCallBackOGC.java

```diff
@@ -0,0 +1,160 @@
+            return format;
+        } else if (accept != null) {
+            if (accept.contains(TemplateIdentifier.GEOJSON.getOutputFormat()))
+                return TemplateIdentifier.GEOJSON.getOutputFormat();
+            else if (accept.contains(TemplateIdentifier.JSONLD.getOutputFormat()))
+                return TemplateIdentifier.JSONLD.getOutputFormat();
+        }
+        return null;
+    }
+
+    private void replaceJsonLdPathWithFilter(
+            String strFilter, String outputFormat, Operation operation) throws Exception {
+        if (strFilter != null && strFilter.indexOf(".") != -1) {
+            FeatureTypeInfo typeInfo = getFeatureType((String) operation.getParameters()[0]);
+            RootBuilder root = configuration.getTemplate(typeInfo, outputFormat);
+            if (root != null) {
+                JsonPathVisitor visitor = new JsonPathVisitor(typeInfo.getFeatureType());
+                /* Todo find a better way to replace json-ld path with corresponding template attribute*/
+                // Get filter from string in order to make it accept the visitor
+                Filter f = (Filter) XCQL.toFilter(strFilter).accept(visitor, root);
+                // Taking back a string from Function cause
+                // OGC API get a string cql filter from query string
+                String newFilter =
+                        JsonLdCQLManager.removeQuotes(ECQL.toCQL(f)).replaceAll("/", ".");
+                newFilter = JsonLdCQLManager.quoteXpathAttribute(newFilter);
+                for (int i = 0; i < operation.getParameters().length; i++) {
+                    Object p = operation.getParameters()[i];
+                    if (p != null && ((String.valueOf(p)).trim().equals(strFilter.trim()))) {
+                        operation.getParameters()[i] = newFilter;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public Response responseDispatched(
+            Request request, Operation operation, Object result, Response response) {
+        String format = getFormatSupportingTemplating(request);
+        if (format != null && format.equals(TemplateIdentifier.GEOJSON.getOutputFormat())) {
+            FeatureTypeInfo typeInfo = null;
+            typeInfo = getFeatureType((String) operation.getParameters()[0]);
+            if (typeInfo != null) {
+                try {
+                    RootBuilder root = configuration.getTemplate(typeInfo, format);
+                    if (root != null) {
+                        GeoJSONGetFeatureResponse featureResponse =
+                                (GeoJSONGetFeatureResponse)
+                                        GeoServerExtensions.bean("geoJSONGetFeatureResponse");
+                        GeoJsonTemplateGetFeatureResponse templatingResp =
+                                new GeoJsonTemplateGetFeatureResponse(
```

Comment:
Could you please reduce a bit the nesting level? There is no requirement to put everything in the same method, once you determined you want to replace the response, you can call a different method that does exactly that, and give it a name that explains the purpose in the process too (e.g., "replaceResponse(...)").

Reply:
fixed